### PR TITLE
ARROW-710: [Python] Read/write with file-like Python objects from read_feather/write_feather

### PR DESCRIPTION
**Pull request by wesm**
cc @jreback

### CODE REVIEW

**Comment by jreback** [comment]
very nice!

File: cpp/src/arrow/ipc/feather.cc [hunk 1]
```diff
@@ -517,9 +511,8 @@ class TableWriter::TableWriterImpl : public ArrayVisitor {
     // Footer: metadata length, magic bytes
     RETURN_NOT_OK(
         stream_->Write(reinterpret_cast<const uint8_t*>(&buffer_size), sizeof(uint32_t)));
-    RETURN_NOT_OK(stream_->Write(reinterpret_cast<const uint8_t*>(kFeatherMagicBytes),
-        strlen(kFeatherMagicBytes)));
-    return stream_->Close();
+    return stream_->Write(
```
Comment (cpcloud) [inline cpp/src/arrow/ipc/feather.cc]: Should this also be `RETURN_NOT_OK(...)`?

Comment (cpcloud) [inline cpp/src/arrow/ipc/feather.cc]: Ah, nevermind.

**Review by cpcloud — APPROVED** [review]
LGTM.

**Pull request closed without merging**